graphql_api/tests/test_current_user_ariadne.py

```diff
@@ -179,7 +179,7 @@ def test_fetch_terms_agreement_and_business_email_when_owner_profile_is_null(sel
         }
         """
         data = self.gql_request(query, owner=current_user)
-        assert data == {"me": {"businessEmail": None, "termsAgreement": None}}
+        assert data == {"me": {"businessEmail": None, "termsAgreement": False}}
```

Comment:
Why did this value go from None to False?

For termsAgreement here:
- True means they accepted our terms
- None means they haven't agreed/disagreed with our terms (they haven't interacted with the forms)
- False means they declined our terms

So there's an important distinction here. I think this might be a side-effect of your factory defaulting to False instead of None.

Reply:
I set agreement to False by default when a user is created because it follows the logic of setting agreement in owner profile to False when an Owner is created.
https://github.com/codecov/codecov-api/blob/ab2eaaac8e0a007b6e1232b79a9fa1df0522b05c/codecov_auth/signals.py#L17

Reply:
I believe though since we're moving from profile to user here, then the test for this should be adapted to deleting the user instead of the owner profile. I think it's odd that the logic is meant to "remain the same" so our test resutls shouldn't change, but they are here 👀